p2p/transport.go

```diff
@@ -207,7 +207,11 @@ func (mt *MultiplexTransport) Dial(
 func (mt *MultiplexTransport) Close() error {
 	close(mt.closec)
 
-	return mt.listener.Close()
+	if mt.listener != nil {
```

Comment:
Is this to not panic when `Close` is called before `Listen`? When does that happen?

Reply:
Yes. There's a problem with the current common#Service implementation where it considers a service started before OnStart() finishes. https://github.com/tendermint/tendermint/blob/0c9c3292c918617624f6f3fbcd95eceade18bcd5/libs/common/service.go#L131-L139

Therefore, calling IsRunning() can return true even though the service is not yet started (or partially started)